src/WebJobs.Extensions.CosmosDB/Trigger/CosmosDBTriggerAttribute.cs

```diff
@@ -89,5 +90,26 @@ public CosmosDBTriggerAttribute(string databaseName, string collectionName)
         /// collection.
         /// </summary>
         public int LeasesCollectionThroughput { get; set; }
+
+        /// <summary>
+        /// Optional, defaults to false.
+        /// If false, any exception thrown by the function will be ignored and the checkpoint updated.
+        /// If true, throwing an exception from the function will cause a series of retries with back-off.  If those also fail, 
+        /// the Feed Processor will not checkpointed and processing will be halted for the 
+        /// current partition for a brief time, after which any notifications since the last checkpoint will be re-delivered.
+        /// When set to true, the function should implement its own any poison document handling logic in order to prevent an un-processable notification from 
+        /// permanently halting processing of the partition and potentially running up execution costs; this flag is meant to enable guaranteed
+        /// processing of every notification, even in case of a temporary "downstream dependency failure" preventing the function from completing
+        /// successfully.
+        /// </summary>
+        public bool RetryOnFailure { get; set; } = false;
+
+        /// <summary>
+        /// Optional, defaults to Incremental.DefaultExponential
+        /// This setting takes effect when RetryOnFailure is set to true (it defaults to false) and controls the local 
+        /// retry strategy used before abandoning the checkpoint.  
+        /// For an example of a custom retry strategy, see https://msdn.microsoft.com/en-us/library/hh680943(v=pandp.50).aspx
+        /// </summary>
+        public RetryStrategy RetryStrategy { get; set; } = Incremental.DefaultExponential;
```

Comment:
I don't think this will work as an Attribute property since C# requires that an attribute property be a compile-time constant value. @ealsur just hit this with another property.

Can we simplify this down to be something like `MaxRetryCount` and handle all the details internally? The default could be 0 to keep current behavior. We'd want this metadata to be easily definable in JSON for Functions -- so keeping the property simpler would benefit us there as well.

Reply:
The issue that Brett refers is that the JIT seems to only allow primitives to be passed as parameters. It will compile but fail at runtime. I tried it before 😢 
So go for the suggestion that Brett stated, it will save you headaches 😄

Reply:
Thanks for the heads up!  Per discussion I have changed this to an int value.